contracts/modules/utils/RequireUtils.sol

```diff
@@ -61,11 +77,100 @@ contract RequireUtils {
             )
           )
         )
-      ) == _wallet, "RequireUtils#requireConfig: UNEXPECTED_COUNTERFACTUAL_IMAGE_HASH");
+      ) == _wallet, "RequireUtils#publishConfig: UNEXPECTED_COUNTERFACTUAL_IMAGE_HASH");
     }
 
     // Emit event for easy config retrieval
     emit RequiredConfig(_wallet, imageHash, _threshold, abi.encode(_members));
+
+    if (_index) {
+      // Register last event for given wallet
+      lastWalletUpdate[_wallet] = block.number;
+    }
+  }
+
+  function publishSigners(
```

Comment:
can you add a comment above this method like:

```
// publishSigners records the initial wallet config. It should only be called to record the initial set of signers, and not called between wallet updates.
```

+ maybe a few more details? so its clear how this method should be used

Reply:
I think the comment is outdated

Reply:
Yep, I can add some comments.

The method can be used in any point in the life-cycle of the wallet, including after updating and deploying. It takes a signature and it validates that each signer belongs to the wallet, constructs the configuration and it also validates the configuration is on the wallet (as `publishConfig`).

This can be used to later retrieve which wallets a given signer participates, it validates the signature to avoid a possible attacker creating 1000s of fake wallets and making the task of finding the right one difficult.

In practice, we are going to use this on the wallet webapp to skip the counter-factual generation of the initial session key, instead thanks to this method we can find the wallet address even if it's not 100% counterfactual, but looking all the Sequence wallets a signer is a part of.